build_tools/hack/build_env_diag.py

```diff
@@ -18,1123 +19,2202 @@
+                "warn", cmake, f"Found CMake {cmake.MAJOR_VERSION} at {cmake.exe}"
+            )
+            _except = cstring(
                 f"""
-    {Emoji_Warning} We detected your Visual Studio version is {VISUAL_STUDIO_VERSION}.
-    We may consider your Visual Studio version and its toolchain could be too old and may not have official support
-    after TheRock official release.
-    Please consider upgrade your Visual Studio version and toolchain.
+    We found you're using CMake program is CMake 4 (cmake {cmake.version}).
+    The support of CMake 4 is still not confirmed, and the different CMake behavior may effect TheRock build.
+    Please downgrade it and re-try again.
+        traceback: CMake program too new may cause unstable
+            expected: 3.25.X ≤ cmake ≤ 3.31.X, found {cmake.version}
+    """,
+                "warn",
+            )
+            _result = False
 
-    traceback: Visual Studio version {VISUAL_STUDIO_VERSION} too old
+        elif cmake.MAJOR_VERSION == 3 and cmake.MINOR_VERSION < 25:
+            _stat = self.msg_stat(
+                "warn", cmake, f"Found CMake {cmake.version} at {cmake.exe}"
+            )
+            _except = cstring(
+                f"""
+    We found you're CMake program is ({cmake.version}).
+    Your CMake version is too old to TheRock project that requires version 3.25.
+    Please upgrade your CMake program version.
+        traceback: CMake program too old excluded by TheRock Top-Level CMakeLists.txt rules `cmake_minimum_required()`
+            expected: 3.25.X ≤ cmake ≤ 3.31.X, found {cmake.version}
     """,
-                Color_Warning,
+                "warn",
             )
+            _result = False
 
-    else:
-        print(f"{Emoji_Error} [{program}] {program} environment not detected. ")
-        printc(
-            f"""
-    We detected you're running out of Visual Studio developer environment.
-    MSVC compilers needs 'Developer prompt/PowerShell for Visual Studio 20XX' to load its tools and libraries.
-    Please re-run a terminal or load ENV variable batch files to add MSVC toolchain.
+        else:
+            _stat = self.msg_stat(
+                "pass", cmake, f"Found CMake {cmake.version} at {cmake.exe}"
+            )
+            _except = ""
+            _result = True
+
+        return _stat, _except, _result
 
-    traceback: Developer for {program} ENV not detected/activated
+    def check_ninja(self):
+        ninja = self.device.ninja
+        ninja.description = "A small buildsystem focus on speed"
+
+        if ninja.exe is None:
+            _stat = self.msg_stat("err", ninja, f"Ninja Generator not found.")
+            _except = cstring(
+                f"""
+    We can't find required generator "Ninja".
+    Ninja is TheRock project current supported generator.
+    For Windows users, please use ninja from VS20XX or Strawberry Perl, or build from source. Install from command line, please avoid version 1.11.
```

Comment:
Let's not recommend Strawberry Perl for anything but gfortran (here or elsewhere)